sale_order_project/wizard/sale_order_createproject.py

```diff
@@ -0,0 +1,42 @@
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class SaleOrderCreateProject(models.TransientModel):
+    """ wizard to create a Project from a Sale Order """
+    _name = "sale.order.createproject"
+
+    sale_order_id = fields.Many2one('sale.order', string='Order', domain=[('type', '=', 'order')])
+    related_project_id = fields.Many2one(
+        'project.project',
+        string='Project',
+        help=_("Leave it blank if you want create a new project with the sale order's name as default name.")
+    )
+
+    @api.multi
+    def action_create_project_task(self):
+        self.ensure_one()
+        # get the order to update
+        order = self.sale_order_id
+        project = self.related_project_id
+
+        # if related_project_id is empty
+        if not project.id and not order.related_project_id:
+            # create new project.project
+            order.action_create_project()
+        # if project but order is not related to that project
```

Comment:
IMO Remove dead code everywhere

Reply:
Which dead code? they are comments, can you explain me it, please? :D

Reply:
Oh, ok, i'm deleting it right now.